tgui/packages/tgui/interfaces/PersonalCooking.tsx

```diff
@@ -0,0 +1,579 @@
+import { BooleanLike, classes } from 'common/react';
+import { createSearch } from 'common/string';
+import { flow } from 'common/fp';
+import { map, filter, sortBy } from 'common/collections';
+import { useBackend, useLocalState } from '../backend';
+import { Divider, Button, Section, Tabs, Stack, Box, Input, NoticeBox, Icon, Tooltip } from '../components';
+import { Window } from '../layouts';
+import { Food } from './PreferencesMenu/data';
+
+const TYPE_ICONS = {
+  'Can Make': 'utensils',
+  [Food.Alcohol]: 'wine-glass',
+  [Food.Breakfast]: 'sun',
+  [Food.Bugs]: 'bug',
+  [Food.Cloth]: 'tshirt',
+  [Food.Dairy]: 'cheese',
+  [Food.Fried]: 'fire',
+  [Food.Fruit]: 'apple-alt',
+  [Food.Gore]: 'skull',
+  [Food.Grain]: 'wheat-awn',
+  [Food.Gross]: 'trash',
+  [Food.Junkfood]: 'pizza-slice',
+  [Food.Meat]: 'bacon',
+  [Food.Nuts]: 'seedling',
+  [Food.Oranges]: 'apple-alt',
+  [Food.Pineapple]: 'apple-alt',
+  [Food.Raw]: 'drumstick-bite',
+  [Food.Seafood]: 'fish',
+  [Food.Sugar]: 'candy-cane',
+  [Food.Toxic]: 'biohazard',
+  [Food.Vegetables]: 'carrot',
+} as const;
+
+const CATEGORY_ICONS = {
+  'Can Make': 'utensils',
+  'Breads': 'bread-slice',
+  'Burgers': 'burger',
+  'Cakes': 'cake-candles',
+  'Egg-Based Food': 'egg',
+  'Frozen': 'ice-cream',
+  'Lizard Food': 'dragon',
+  'Meats': 'bacon',
+  'Mexican Food': 'pepper-hot',
+  'Misc. Food': 'bowl-food',
+  'Mothic Food': 'shirt',
+  'Pastries': 'cookie',
+  'Pies': 'chart-pie',
+  'Pizzas': 'pizza-slice',
+  'Salads': 'leaf',
+  'Sandwiches': 'hotdog',
+  'Seafood': 'fish',
+  'Soups': 'mug-saucer',
+  'Spaghettis': 'wheat-awn',
+} as const;
+
+type Ingredient = {
+  path: string;
+  name: string;
+  amount: number;
+  is_reagent: BooleanLike;
+};
+
+type Recipe = {
+  ref: string;
+  result: string;
+  name: string;
+  desc: string;
+  reqs: Ingredient[];
+  category: string;
+  foodtypes: string[];
+  non_craftable: BooleanLike;
+  is_reaction: BooleanLike;
+  steps: string[];
+  tool_paths: Ingredient[];
+  catalysts: Ingredient[];
+  machinery: Ingredient[];
+};
+
+type Diet = {
+  liked_food: string[];
+  disliked_food: string[];
+  toxic_food: string[];
+};
+
+type Data = {
+  // Dynamic
+  busy: BooleanLike;
+  display_compact: BooleanLike;
+  display_craftable_only: BooleanLike;
+  craftability: Record<string, BooleanLike>;
+  // Static
+  recipes: Recipe[];
+  categories: string[];
+  foodtypes: string[];
+  diet: Diet;
+};
+
+export const PersonalCooking = (props, context) => {
+  const {
+    act,
+    data: {
+      busy,
+      display_compact,
+      display_craftable_only,
+      craftability,
+      recipes,
```

Comment:
Hey, I think I recommended you to not extract variables from data in your cooking ui, because they're only used to prepare the data and are never used again, and now you're stuck with names that have `_combined` and `_filtered` appended to them, and they're used way more. I understand that maint asked you to do this, but it's poor taste. Variable names that are used more than once must be short and clear and transitional data should ideally never pollute the scope.

You can keep things like `busy`, `display_compact`, etc, because this is not transitional data.

Also note that we prefer destructuring things like so, because it takes less space and allows accessing data directly for performing later transformations on the data.

```tsx
const { act, data } = useBackend<Data>(context);
const { x, y, z } = data;
```